Deployment from nightly failing when path is used for first time

```
(.venv) [~/git/ocs-ci]$ run-ci -m deployment -vrA --no-print-logs --cluster-conf conf/ocs_basic_install.yml --cluster-name ratamir-cluster2 --log-level DEBUG --cluster-path /home/ratamir/ocs_cluster2 --capture=no tests/ --deploy
The kubeconfig file /home/ratamir/ocs_cluster2/auth/kubeconfig doesn't exist!
==================================================================================== test session starts ====================================================================================
platform linux -- Python 3.7.3, pytest-4.5.0, py-1.8.0, pluggy-0.11.0 -- /home/ratamir/git/ocs-ci/.venv/bin/python
cachedir: .pytest_cache
metadata: {'Python': '3.7.3', 'Platform': 'Linux-5.0.16-100.fc28.x86_64-x86_64-with-fedora-28-Twenty_Eight', 'Packages': {'pytest': '4.5.0', 'py': '1.8.0', 'pluggy': '0.11.0'}, 'Plugins': {'reportportal': '1.0.5', 'cov': '2.7.1', 'metadata': '1.8.0', 'marker-bugzilla': '0.9.1.dev2', 'logger': '0.5.0', 'html': '1.21.1'}}
rootdir: /home/ratamir/git/ocs-ci, inifile: pytest.ini
plugins: reportportal-1.0.5, cov-2.7.1, metadata-1.8.0, marker-bugzilla-0.9.1.dev2, logger-0.5.0, html-1.21.1
collected 92 items / 91 deselected / 1 selected                                                                                                                                             

tests/ecosystem/deployment/test_ocs_basic_install.py::test_cluster_is_running 
-------------------------------------------------------------------------------------- live log setup ---------------------------------------------------------------------------------------
13:03:57 - MainThread - tests.conftest - INFO - All logs located at /tmp/ocs-ci-logs-1565863436
13:03:57 - MainThread - ocs_ci.deployment.factory - INFO - Deployment key = aws_ipi
13:03:57 - MainThread - ocs_ci.deployment.factory - INFO - Current deployment platform: aws,deployment type: ipi
13:03:58 - MainThread - ocs_ci.utility.utils - INFO - Downloading openshift client (4.2.0-0.nightly-2019-08-15-073735).
ERROR

========================================================================================== ERRORS ===========================================================================================
_________________________________________________________________________ ERROR at setup of test_cluster_is_running _________________________________________________________________________

request = <SubRequest 'cluster' for <Function test_cluster_is_running>>, log_cli_level = 'INFO'
```

    @pytest.fixture(scope="session", autouse=True)
    def cluster(request, log_cli_level):
        """
        This fixture initiates deployment for both OCP and OCS clusters.
        Specific platform deployment classes will handle the fine details
        of action
        """
        log.info(f"All logs located at {ocsci_log_path()}")
    
        teardown = config.RUN['cli_params']['teardown']
        deploy = config.RUN['cli_params']['deploy']
        factory = dep_factory.DeploymentFactory()
        deployer = factory.get_deployment()
    
        # Add a finalizer to teardown the cluster after test execution is finished
        if teardown:
            def cluster_teardown_finalizer():
                deployer.destroy_cluster(log_cli_level)
            request.addfinalizer(cluster_teardown_finalizer)
            log.info("Will teardown cluster because --teardown was provided")
    
        # Download client
        force_download = (
            config.RUN['cli_params'].get('deploy')
            and config.DEPLOYMENT['force_download_client']
        )
>       get_openshift_client(force_download=force_download)

../tests/conftest.py:404: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../ocs_ci/utility/utils.py:582: in get_openshift_client
    download_file(url, tarball)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

url = 'https://openshift-release-artifacts.svc.ci.openshift.org/4.2.0-0.nightly-2019-08-15-073735/openshift-client-linux-4.2.0-0.nightly-2019-08-15-073735.tar.gz'
filename = 'openshift-client.tar.gz'

    def download_file(url, filename):
        """
        Download a file from a specified url
    
        Args:
            url (str): URL of the file to download
            filename (str): Name of the file to write the download to
    
        """
        log.debug(f"Download '{url}' to '{filename}'.")
        with open(filename, "wb") as f:
            r = requests.get(url)
            f.write(r.content)
>       assert r.ok
E       AssertionError

../ocs_ci/utility/utils.py:402: AssertionError
================================================================================== short test summary info ==================================================================================
ERROR tests/ecosystem/deployment/test_ocs_basic_install.py::test_cluster_is_running - AssertionError
========================================================================== 91 deselected, 1 error in 1.78 seconds ===========================================================================

--- a/ocs_ci/ocs/exceptions.py
+++ b/ocs_ci/ocs/exceptions.py
@@ -42,3 +42,7 @@ class MDSCountException(Exception):
 
 class DeploymentPlatformNotSupported(Exception):
     pass
+
+
+class UnavailableBuildException(Exception):
+    pass

--- a/ocs_ci/utility/utils.py
+++ b/ocs_ci/utility/utils.py
@@ -16,6 +16,7 @@ import smtplib
 from ocs_ci.ocs.exceptions import CephHealthException
 from ocs_ci.ocs.exceptions import (
     CommandFailed, UnsupportedOSType, TimeoutExpiredError,
+    UnavailableBuildException,
 )
 from ocs_ci.framework import config
 from ocs_ci.utility.aws import AWS
@@ -398,8 +399,10 @@ def download_file(url, filename):
     log.debug(f"Download '{url}' to '{filename}'.")
     with open(filename, "wb") as f:
         r = requests.get(url)
+        assert r.ok, (
+            f"The URL {url} is not available! Status: {r.status_code}."
+        )
         f.write(r.content)
-    assert r.ok
 
 
 def get_url_content(url):
@@ -417,7 +420,7 @@ def get_url_content(url):
     """
     log.debug(f"Download '{url}' content.")
     r = requests.get(url)
-    assert r.ok, f"Couldn't load URL: {url} content!"
+    assert r.ok, f"Couldn't load URL: {url} content! Status: {r.status_code}."
     return r.content
 
 
@@ -591,6 +594,15 @@ def get_openshift_client(
     return client_binary_path
 
 
+def ensure_nightly_build_availability(build_url):
+    base_build_url = build_url.rsplit('/', 1)[0]
+    r = requests.get(base_build_url)
+    extracting_condition = b"Extracting" in r.content
+    if extracting_condition:
+        log.info("Build is extracting now, may take up to a minute.")
+    return r.ok and not extracting_condition
+
+
 def get_openshift_mirror_url(file_name, version):
     """
     Format url to OpenShift mirror (for client and installer download).
@@ -602,6 +614,9 @@ def get_openshift_mirror_url(file_name, version):
     Returns:
         str: Url of the desired file (installer or client)
 
+    Raises:
+        UnsupportedOSType: In case the OS type is not supported
+        UnavailableBuildException: In case the build url is not reachable
     """
     if platform.system() == "Darwin":
         os_type = "mac"
@@ -613,6 +628,14 @@ def get_openshift_mirror_url(file_name, version):
         f"https://openshift-release-artifacts.svc.ci.openshift.org/"
         f"{version}/{file_name}-{os_type}-{version}.tar.gz"
     )
+    sample = TimeoutSampler(
+        timeout=60, sleep=5, func=ensure_nightly_build_availability,
+        build_url=url,
+    )
+    if not sample.wait_for_func_status(result=True):
+        raise UnavailableBuildException(
+            f"The build url {url} is not reachable"
+        )
     return url
 
 
@@ -719,7 +742,7 @@ class TimeoutSampler(object):
                 timeout=60, sleep=1, func=some_func, func_arg1="1",
                 func_arg2="2"
             )
-            if not sample.waitForFuncStatus(result=True):
+            if not sample.wait_for_func_status(result=True):
                 raise Exception
         """
         try:

